code/game/objects/effects/fluids.dm

```diff
@@ -116,6 +96,39 @@
 		else
 			set_light(0)
 
+/obj/effect/fluid/proc/can_burn()
+	if(reagents?.total_volume && reagents.total_volume < FLUID_DEEP)
+		for(var/rtype in reagents.reagent_volumes)
+			var/decl/material/mat = GET_DECL(rtype)
+			if(mat.fuel_value)
+				return TRUE
+	return FALSE
+
+/obj/effect/fluid/proc/burn()
+
+	if(!reagents || !loc)
+		return 0
+
+	var/update_air = FALSE
+	var/update_reagents = FALSE
+	var/datum/gas_mixture/air_contents = loc.return_air()
+	for(var/rtype in reagents?.reagent_volumes)
+		var/decl/material/mat = GET_DECL(rtype)
+		var/fuel_amount = reagents.reagent_volumes[rtype] * FIRE_LIQUD_MIN_BURNRATE
+		if(fuel_amount)
```

Comment:
wouldn't this make non-burnabe reagents disappear too, since you only check fuel_value inside of this block for return value?

Reply:
Oh yeah.